pkgs/development/python-modules/unidic/default.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, mecab
+, setuptools-scm
+, requests
+, tqdm
+, wasabi
+, plac
+, cython
+, platformdirs
+}:
+
+buildPythonPackage rec {
+  pname = "unidic";
+  version = "1.1.0";
+  format = "setuptools";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "polm";
+    repo = "unidic-py";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-srhQDXGgoIMhYuCbyQB3kF4LrODnoOqLbjBQMvhPieY=";
+  };
+
+  postPatch = ''
+    substituteInPlace requirements.txt \
+      --replace "wasabi>=0.6.0,<1.0.0" "wasabi"
+    substituteInPlace setup.cfg \
+      --replace "wasabi>=0.6.0,<1.0.0" "wasabi"
+  '';
+
+  # no tests
+  doCheck = false;
+
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  propagatedBuildInputs = [ requests tqdm wasabi plac platformdirs ];
+
+  nativeBuildInputs = [ cython mecab setuptools-scm ];
+
+  pythonImportsCheck = [ "unidic" ];
+
+  patches = [ ./download.patch ];
```

Comment:
What is that patch doing? also patches goes between postPatch and src

Reply:
The library needs to download a dictionary from the internet at runtime. It tries to download the file where the library is located (in the nix store, which is read-only) and crashes. This patch changes the download directory to the user's home .cache folder. Though as @natsukium said we could also pick another destination folder.

Reply:
Sorry for the late reply. The purpose of the patch is as @Laurent2916 described above.
I found out that there is a way for users to download their own dictionaries and use them in scripts, so the current patch looks fine to me now.